modules/miner/splitsetheap_test.go

```diff
@@ -0,0 +1,450 @@
+		}
+
+	}
+}
+
+// TestMapHeapRemoveBySetID pushes a sequence of elements onto
+// a max-heap and min-heap. Then it removes a random element using
+// its splitSetID, and checks that it has been removed.
+func TestMapHeapRemoveBySetID(t *testing.T) {
+	max := &mapHeap{
+		selectID: make(map[splitSetID]*mapElement),
+		data:     make([]*mapElement, 0),
+		size:     0,
+		minHeap:  false,
+	}
+
+	min := &mapHeap{
+		selectID: make(map[splitSetID]*mapElement),
+		data:     make([]*mapElement, 0),
+		size:     0,
+		minHeap:  true,
+	}
+
+	max.Init()
+	min.Init()
+
+	for i := 0; i < 5000; i++ {
+		e1 := &mapElement{
+			set: &splitSet{
+				averageFee:   types.SiacoinPrecision.Mul64(uint64(i)),
+				size:         uint64(10 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		e2 := &mapElement{
+			set: &splitSet{
+				averageFee:   types.SiacoinPrecision.Mul64(uint64(i)),
+				size:         uint64(10 * i),
+				transactions: make([]types.Transaction, 0),
+			},
+
+			id:    splitSetID(i),
+			index: 0,
+		}
+		max.Push(e1)
+		min.Push(e2)
+	}
+
+	randID := splitSetID(rand.Intn(5000))
+	firstToBeRemoved := max.selectID[randID]
+
+	// Iterate over data in min heap and max heap to confirm the element to be
+	// removed is actually there.
+	inMaxHeap := false
+	inMinHeap := false
+	for _, v := range max.data {
+		if v.id == firstToBeRemoved.id {
+			t.Log(v)
+			inMaxHeap = true
+			break
+		}
+	}
+	for _, v := range min.data {
+		if v.id == firstToBeRemoved.id {
+			t.Log(v)
+			inMinHeap = true
+			break
+		}
+	}
+
+	if !inMinHeap || !inMaxHeap {
+		t.Error("Element not found in heap(s) before being removed by splitSetID.")
+	}
+
+	if max.selectID[randID] == nil || min.selectID[randID] == nil {
+		t.Error("Element not found in map(s) before being removed by splitSetID")
+	}
+
+	max.RemoveSetByID(randID)
+	min.RemoveSetByID(randID)
+
+	// Iterate over data in min heap and max heap to confirm the element to be
+	// removed was actually removed
+	removedFromMax := true
+	removedFromMin := true
+	for _, v := range max.data {
+		if v.id == firstToBeRemoved.id {
+			removedFromMax = false
+			break
+		}
+	}
+	for _, v := range min.data {
+		if v.id == firstToBeRemoved.id {
+			removedFromMin = false
+			break
+		}
+	}
+
+	if !removedFromMin {
+		t.Error("Element found in  min heap(s) after being removed by splitSetID.")
+	}
+
+	if !removedFromMax {
+		t.Error("Element found in  max heap(s) after being removed by splitSetID.")
+	}
+
+	_, inMinMap := min.selectID[randID]
+	_, inMaxMap := max.selectID[randID]
+	if inMinMap {
+		t.Error("Element found in min map(s) after being removed by splitSetID")
+	}
+	if inMaxMap {
+		t.Error("Element found in max map(s) after being removed by splitSetID")
+	}
+
+}
+
+// TestMapHeapPeek test the Peek method. First, on an empty heap Peek should
+// return false. Then it checks that Peek returns the same result as the next Pop.
+func TestMapHeapPeek(t *testing.T) {
```

Comment:
same here, should be pretty easy to tag on a check that the heap size does not change from a peek.